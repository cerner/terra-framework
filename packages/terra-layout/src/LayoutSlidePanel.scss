:local {
  $hover-target-width: var(--layout-slide-hover-target-width, 2.143rem);

  .layout-slide-panel {
    height: 100%;
    overflow: hidden;
    position: relative;
    width: 100%;
  }

  .content,
  .panel {
    box-sizing: border-box;
    height: 100%;
    overflow: hidden;
    transition: none;
  }

  .is-animated {
    transition: transform 0.15s ease;
  }

  .content {
    float: right;
    position: relative;
    width: 100%;
    z-index: 0;
  }

  .panel {
    background-color: var(--layout-slide-panel-background-color, #fff);
    position: absolute;
    transform: translate3d(-100%, 0, 0);
    width: var(--layout-slide-panel-desktop-panel-width, 18.571rem);
    z-index: 5000;
  }

  // Move the slider back to its origin when its on-screen.
  .is-open > .panel {
    transform: translate3d(0, 0, 0);
  }

  .hover-toggle-enabled.is-overlay {
    > .content {
      width: calc(100% - #{$hover-target-width});// allow for hover bar
    }

    > .panel {
      transform: translateX(-100%) translateX($hover-target-width);
    }

    &.is-open > .panel {
      transform: translate3d(0, 0, 0);
    }
  }

  // Due to conflicts with the RTL plugin and the use of the variable, we need to explicitly
  // specify the rtl-based transform value.
  /* stylelint-disable selector-max-compound-selectors */
  [dir=rtl] {
    .hover-toggle-enabled.is-overlay {
      > .panel {
        transform: translateX(100%) translateX(calc(-1 * #{$hover-target-width}));
      }
    }
  }
  /* stylelint-enable */

  .panel.is-tiny {
    width: var(--layout-slide-panel-phone-panel-width, calc(100% - 3.571rem));
  }

  .panel.is-small {
    width: var(--layout-slide-panel-tablet-panel-width, 22.857rem);
  }

  // Overlay
  .is-open.is-overlay > .panel {
    box-shadow: 0.07143rem 0 0.3571rem var(--layout-slide-panel-panel-shadow-color, #bcbfc0);
  }

  // Squish
  .is-open.is-squish {
    > .content {
      width: calc(100% - #{var(--layout-slide-panel-desktop-panel-width, 18.571rem)} - 0.07143rem);
    }

    > .panel {
      border-right: 0.07143rem solid var(--layout-slide-panel-panel-border-color, #d3d4d5);
      transition: none;
      width: calc(0.07143rem + #{var(--layout-slide-panel-desktop-panel-width, 18.571rem)});
    }
  }
}
