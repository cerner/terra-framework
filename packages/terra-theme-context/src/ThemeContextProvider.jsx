import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import ThemeContext from './ThemeContext';

const propTypes = {
  /**
   * The components to be rendered within the context of the ThemeContextProvider. Components rendered here are able to interact with ThemeContextProvider through the ThemeContext.
   */
  children: PropTypes.element.isRequired,
  /**
   * An object containing the name and className of the selected theme.
   */
  theme: PropTypes.shape({
    name: PropTypes.string,
    className: PropTypes.string,
  }),
};

const defaultProps = {
  theme: {},
};

const ThemeContextProvider = ({ theme, children }) => {
  const contextValue = useMemo(() => ({ name: theme.name, className: theme.className }), [theme.name, theme.className]);
  return (
    <ThemeContext.Provider value={contextValue}>
      {children}
    </ThemeContext.Provider>
  );
};

ThemeContextProvider.propTypes = propTypes;
ThemeContextProvider.defaultProps = defaultProps;

export default ThemeContextProvider;
