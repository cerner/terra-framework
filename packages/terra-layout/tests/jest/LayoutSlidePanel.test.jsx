import React from 'react';

import ThemeContextProvider from 'terra-theme-context/lib/ThemeContextProvider';
import LayoutSlidePanel from '../../src/_LayoutSlidePanel';

describe('LayoutSlidePanel', () => {
  it('should render without optional props', () => {
    const result = shallow((
      <LayoutSlidePanel
        size="medium"
      />
    ));
    expect(result).toMatchSnapshot();
  });

  it('should render with provided props', () => {
    const result = shallow((
      <LayoutSlidePanel
        isAnimated
        isOpen
        isToggleEnabled
        panelBehavior="overlay"
        panelContent={(
          <div>Panel Content</div>
        )}
        size="medium"
        onToggle={() => {}}
        toggleText="Test Text"
      >
        <div>Child</div>
      </LayoutSlidePanel>
    ));

    expect(result).toMatchSnapshot();
  });

  it('should render when small', () => {
    const result = shallow((
      <LayoutSlidePanel
        isAnimated
        isOpen
        isToggleEnabled
        panelBehavior="overlay"
        panelContent={(
          <div>Panel Content</div>
        )}
        size="small"
        onToggle={() => {}}
        toggleText="Test Text"
      >
        <div>Child</div>
      </LayoutSlidePanel>
    ));

    expect(result).toMatchSnapshot();
  });

  it('should render when tiny', () => {
    const result = shallow((
      <LayoutSlidePanel
        isAnimated
        isOpen
        isToggleEnabled
        panelBehavior="overlay"
        panelContent={(
          <div>Panel Content</div>
        )}
        size="tiny"
        onToggle={() => {}}
        toggleText="Test Text"
      >
        <div>Child</div>
      </LayoutSlidePanel>
    ));

    expect(result).toMatchSnapshot();
  });

  it('correctly applies the theme context className', () => {
    const layout = mount(
      <ThemeContextProvider theme={{ className: 'orion-fusion-theme' }}>
        <LayoutSlidePanel
          size="medium"
        />
      </ThemeContextProvider>,
    );
    expect(layout).toMatchSnapshot();
  });
});
