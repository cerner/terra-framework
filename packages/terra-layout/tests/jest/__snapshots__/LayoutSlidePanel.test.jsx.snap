// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutSlidePanel should render when small 1`] = `
<div
  className="layout-slide-panel is-open is-overlay"
>
  <div
    aria-hidden="false"
    className="panel is-small is-animated"
  >
    <div>
      Panel Content
    </div>
  </div>
  <OverlayContainer
    className="content"
    overlay={null}
  >
    <Overlay
      backgroundStyle="dark"
      isOpen={true}
      isRelativeToContainer={true}
      isScrollable={false}
      onRequestClose={[Function]}
      rootSelector="#root"
      zIndex="6000"
    />
    <main
      className="main-container"
      data-terra-layout-main={true}
      role="main"
    >
      <div>
        Child
      </div>
    </main>
  </OverlayContainer>
</div>
`;

exports[`LayoutSlidePanel should render when tiny 1`] = `
<div
  className="layout-slide-panel is-open is-overlay"
>
  <div
    aria-hidden="false"
    className="panel is-tiny is-animated"
  >
    <div>
      Panel Content
    </div>
  </div>
  <OverlayContainer
    className="content"
    overlay={null}
  >
    <Overlay
      backgroundStyle="dark"
      isOpen={true}
      isRelativeToContainer={true}
      isScrollable={false}
      onRequestClose={[Function]}
      rootSelector="#root"
      zIndex="6000"
    />
    <main
      className="main-container"
      data-terra-layout-main={true}
      role="main"
    >
      <div>
        Child
      </div>
    </main>
  </OverlayContainer>
</div>
`;

exports[`LayoutSlidePanel should render with provided props 1`] = `
<div
  className="layout-slide-panel is-open is-overlay"
>
  <div
    aria-hidden="false"
    className="panel is-animated"
  >
    <div>
      Panel Content
    </div>
  </div>
  <OverlayContainer
    className="content"
    overlay={null}
  >
    <Overlay
      backgroundStyle="clear"
      isOpen={true}
      isRelativeToContainer={true}
      isScrollable={false}
      onRequestClose={[Function]}
      rootSelector="#root"
      zIndex="6000"
    />
    <main
      className="main-container"
      data-terra-layout-main={true}
      role="main"
    >
      <div>
        Child
      </div>
    </main>
  </OverlayContainer>
</div>
`;

exports[`LayoutSlidePanel should render without optional props 1`] = `
<div
  className="layout-slide-panel is-overlay"
>
  <div
    aria-hidden="true"
    className="panel"
  />
  <OverlayContainer
    className="content"
    overlay={null}
  >
    <Overlay
      backgroundStyle="clear"
      isOpen={false}
      isRelativeToContainer={true}
      isScrollable={false}
      rootSelector="#root"
      zIndex="6000"
    />
    <main
      className="main-container"
      data-terra-layout-main={true}
      role="main"
    />
  </OverlayContainer>
</div>
`;
